src/orchestrator/orchestrator.rs

```diff
@@ -0,0 +1,191 @@
+    /// 4. The processed data from all executors, collected from `tree_rx`, is aggregated into an `AggregationMerkleSumTree`.
+    /// 6. After processing, executors are terminated to release resources.
+    ///
+    async fn create_aggregation_mst(
+        self,
+        number_executor: usize,
+    ) -> Result<AggregationMerkleSumTree<N_ASSETS, N_BYTES>, Box<dyn std::error::Error>>
+    where
+        [usize; N_ASSETS + 1]: Sized,
+        [usize; 2 * (1 + N_ASSETS)]: Sized,
+    {
+        let entries_per_executor = self.entry_csvs.len() / number_executor;
+
+        // Declare channels for internal communication
+        // One of the channels receives parsed data from entry parser to distribute tasks to executor
+        // while the other channel is used the executors that send a result of the tasks.
+        let mut executors = Vec::new();
+        let mut result_collectors = Vec::new();
+
+        let actual_number_of_workers = min(number_executor, self.entry_csvs.len());
+        for i in 0..actual_number_of_workers {
+            let (entries_tx, mut entries_rx) = mpsc::channel(32);
+            let (tree_tx, tree_rx) = mpsc::channel(32);
+            let executor = self.executor_spawner.spawn_executor().await;
+            result_collectors.push((i, tree_rx));
+
+            // 2. Executor
+            //
+            // Spawn executors that process entries with Worker.
+            //
+            // - Receives 'entries' from [entries_rx] channel.
+            // - Processes 'entries' to build a tree (done by worker).
+            // - Sends the resulting 'tree' back via [tree_tx] channel.
+            //
+            executors.push(tokio::spawn(async move {
+                while let Some(task) = entries_rx.recv().await {
+                    let processed_task = executor
+                        .generate_tree::<N_ASSETS, N_BYTES>(task)
+                        .await
+                        .unwrap();
+                    if tree_tx.send(processed_task).await.is_err() {
+                        break;
+                    }
+                }
+            }));
+
+            // 1. Distributing Tasks
+            //
+            // Spawn a distribution thread that distributes entries to executors
+            //
+            // - Loads CSV file from [csv_file_path].
+            // - Parses CSV file into 'entries'.
+            // - Sends 'entries' to executors via [entries_tx] channel.
+            //
+            let (start, end) = self.calculate_task_range(i, number_executor);
+            let entry_csvs_slice = self.entry_csvs[start..end].to_vec(); // Clone only the necessary slice
+
+            tokio::spawn(async move {
+                for file_path in entry_csvs_slice.iter() {
+                    let entries = entry_parser::<_, N_ASSETS, N_BYTES>(file_path).unwrap();
+                    if entries_tx.send(entries).await.is_err() {
+                        break;
```

Comment:
Should there be any error handling?

Reply:
More error handling at [3e39b10](https://github.com/summa-dev/summa-aggregation/pull/3/commits/3e39b105ff25c1ea2d7e99a187c6c59eff62a6c3)